### DISCUSSION TITLE
iperf3 support for tcpreplay

### DISCUSSION THREAD

**Discussion by tgolden313** [esnet/iperf#1437] [Q&A]
I would like to be able to tcpreplay a pcap file as if it were an iperf3 client.  However, the iperf3 server doesn't seem to acknowledge the packets or spit out results.  I've seen webpages showing this working for iperf.  Is there any way to accomplish this?

**Comment by swlars** [comment]
Could you describe your setup? The main issue you might run into would be simulating the control connection with tcpreplay. A packet trace might not work for that.

**Comment by davidBar-On** [comment]
It seems that indeed setting the TCP control connection is the main problem, as @swlars say.  There are two issue:
1. **tcpliveplay** should be used, not tcpreplay.  See [this comment](https://github.com/appneta/tcpreplay/issues/237#issuecomment-284165948).
2. Even with with tcpliveplay establishing the TCP connection fails.  See [this discussion](https://github.com/appneta/tcpreplay/issues/540).  I tried and I have the exact same problem.

I didn't search further whether there are other tools that can perform such replay.
